### PR TITLE
internal/pkg/scaffold/olm-catalog: better CSV version update method

### PR DESCRIPTION
**Pull request by estroz**
**Description of the change:** update CSV versions with a regexp

**Motivation for the change:** the operator name found throughout a CSV's metadata should be found using an exact matching method to prevent incorrect replacements.

### CODE REVIEW

File: internal/pkg/scaffold/olm-catalog/csv.go [hunk 1]
```diff
@@ -188,29 +189,42 @@ func (s *CSV) getCSVPath(ver string) string {
...
+	if csv.GetNamespace() == "" {
+		csv.SetNamespace("placeholder")
+	}
+	if csv.GetAnnotations() == nil {
+		csv.SetAnnotations(map[string]string{})
+	}
+	if caps, ok := csv.GetAnnotations()["capabilities"]; !ok || caps == "" {
+		csv.GetAnnotations()["capabilities"] = "Basic Install"
+	}
+	if csv.Spec.Provider == (olmapiv1alpha1.AppLink{}) {
```
Comment (joelanford) [inline internal/pkg/scaffold/olm-catalog/csv.go]: Nit: are the parens around `olmapiv1alpah1.AppLink{}` necessary?

File: internal/pkg/scaffold/olm-catalog/csv.go [hunk 2]
```diff
@@ -280,28 +294,23 @@ func (s *CSV) updateCSVVersions(csv *olmapiv1alpha1.ClusterServiceVersion) error
...
-	for _, v := range fieldsToUpdate {
-		err := replaceAllBytes(v, []byte(oldVer), []byte(newVer))
-		if err != nil {
-			return err
-		}
-	}
-
-	// Now replace all references to the old operator name.
+	// Replace all references to the old operator name.
 	lowerProjName := strings.ToLower(s.OperatorName)
```
Comment (joelanford) [inline internal/pkg/scaffold/olm-catalog/csv.go]: Nit: `lowerOperatorName`?

File: internal/pkg/scaffold/olm-catalog/csv.go (hunk 1 above)
Comment (estroz) [reply internal/pkg/scaffold/olm-catalog/csv.go]: They are when [comparing](https://stackoverflow.com/a/28447372) struct types directly.

Comment (joelanford) [reply internal/pkg/scaffold/olm-catalog/csv.go]: Ahh, TIL!

**Comment by hasbro17** [comment]
One last nit: Can you update the final commit message to be more specific like in the PR description `update CSV versions with a regexp` instead of the title `better CSV version update method`.

**Comment by estroz** [comment]
@hasbro17 I'll modify the squash message.

**Pull request merged**